src/openrct2/world/Duck.cpp

```diff
@@ -279,12 +278,11 @@ void rct_duck::UpdateFlyAway()
         Invalidate();
 
         int32_t direction = sprite_direction >> 3;
-        int32_t newX = x + (DuckMoveOffset[direction].x * 2);
-        int32_t newY = y + (DuckMoveOffset[direction].y * 2);
-        int32_t newZ = std::min(z + 2, 496);
-        if (map_is_location_valid({ newX, newY }))
+        auto destination = CoordsXYZ{ x + (DuckMoveOffset[direction].x * 2), y + (DuckMoveOffset[direction].y * 2),
+                                      static_cast<int32_t>(std::min(z + 2, 496)) };
```

Comment:
Can you remove the cast if you specify the type in the `std::min<int32_t>`

Reply:
Seems ok, these cast problems are typically warnings that don't show up on `msbuild`, but do on `cmake`, so let's see what the CI says.